### ISSUE TITLE
Closed legend cause problem with form validation

### ISSUE THREAD

**Issue by netzarbeiter** [contao/core#8300]
If a legend with mandatory fields is closed by default there is a problem with form validation.

Legend closed:
![legend_closed](https://cloud.githubusercontent.com/assets/1942550/14275390/4cf4d8c8-fb17-11e5-8bff-ed383537a5e4.png)

Legend opened:
![legend_opened](https://cloud.githubusercontent.com/assets/1942550/14275401/58e9c60c-fb17-11e5-94b7-950507684f4e.png)

**Comment by leofeyer** [comment]
Legends should always be open if they contain mandatory fields. Can the issue be reproduced in the online demo?

**Comment by aschempp** [comment]
Well you can manually close them…

**Comment by netzarbeiter** [comment]
It can be reproduced in the online demo.

- Add a new content element «Hyperlink»
- Close «Hyperlink settings»
- Save

**Comment by netzarbeiter** [comment]
You can reproduce it in the online demo.

- Add a new content element «Hyperlink»
- Close «Hyperlink settings»
- Save

**Comment by leofeyer** [comment]
> Well you can manually close them…

See https://youtu.be/aqAuuSo_p8M?t=90

**Comment by aschempp** [comment]
That only works after reload, which does not happen with browser validation…

**Comment by fritzmg** [comment]
Fieldsets with mandatory fields should be opened via JavaScript before the form is submitted.

**Comment by leofeyer** [comment]
@aschempp Don't know if you have listened, but the video clearly says:

> … also recognizes mandatory fields and makes sure that they are never inside a collapsed section.

**Comment by fritzmg** [comment]
@leofeyer the problem occurs when the user collapses a section manually. The video only states, that sections are always uncollapsed if there are mandatory fields in them when a record is opened. This works as expected. However, this is not what this issue is about.

**Comment by leofeyer** [comment]
@fritzmg The initial description is:

> If a legend with mandatory fields is closed by default

So I don't think it is about manual collapsing. But it actually does not matter, because Contao should **never** collapse a section containing mandatory fields – no matter if the section was closed by default or manually.

**Comment by fritzmg** [comment]
> But it actually does not matter, because Contao should never collapse a section containing mandatory fields – no matter if the section was closed by default or manually.

I don't agree with that. I often collapse sections, regardless of whether they contain mandatory fields or not, while I create or edit a record which has a lot of sections, to get a better overview. Imho it would be detrimental to the user experience if you suddenly disallow the collapse of sections containing mandatory fields.

**Comment by Toflar** [comment]
Yeah, I think opening them before the form is submitted, makes more sense.

**Comment by leofeyer** [comment]
> I often collapse sections, regardless of whether they contain mandatory fields or not

Yes, but only if those fields contain a value. Otherwise collapsing does not make any sense at all, because not filling in the fields will lead to an error message upon submit.

Therefore Contao will always open sections if there are empty mandatory fields.

**Comment by fritzmg** [comment]
> Yes, but only if those fields contain a value. Otherwise collapsing does not make any sense at all, because not filling in the fields will lead to an error message upon submit.

No, I meant I collapse all fields and open them one by one while creating/editing a record. Just because I collapse a section doesn't mean I do not intend to enter any values in the contained fields.

> Therefore Contao will always open sections if there are empty mandatory fields.

Do you mean when loading an entry? Yes, that currently works as expected and does not need to be changed. I was concerned you want to remove the functionality to be able to collapse sections containing mandatory fields altogether.

The issue here only arises if you manually collapse a section containing a mandatory field and then click any of the `[type="submit"]` buttons. Thus the solution would be to simply do something like this (jQuery version):
```JavaScript
$('.tl_submit_container [type="submit"]').click(function(){
	// un-collapse all sections containing mandatory fields here
});
```

**Comment by leofeyer** [comment]
Fixed in 36fbc4142cba749e246c4de27f17e16530a50401.
